lib/azure/armrest/virtual_machine_service.rb

```diff
@@ -116,15 +113,20 @@ def list(group = nil)
       # * overwrite - Boolean that indicates whether or not to overwrite any VHD's
       #               with the same prefix. The default is false.
       #
-      # The :prefix and :container options are mandatory.
+      # The :prefix and :container options are mandatory. You may optionally
+      # specify the :resource_group as an option, or as the 3rd argument, but
+      # it is also mandatory if not already set in the constructor.
       #
       # Note that this is a long running operation. You are expected to
       # poll on the client side to check the status of the operation.
       #
-      def capture(vmname, group = vmname, options = {})
+      def capture(vmname, options = {}, group = armrest_configuration.resource_group)
         prefix = options.fetch(:prefix)
         container = options.fetch(:container)
         overwrite = options[:overwrite] || false
+        group ||= options[:resource_group]
```

Comment:
which group to use? I think group other then `armrest_configuration.resource_group` is the first order, then from `options`, finally, `armrest_configuration.resource_group`.

So, let's set `group = nil` in the argument list, and set it to `armrest_configuration.resource_group` in the body as the last resort.